contracts/modules/token/TokenController.sol

```diff
@@ -19,18 +19,56 @@ import "@openzeppelin/contracts/math/SafeMath.sol";
 import "../../abstract/Iupgradable.sol";
 import "../../interfaces/IPooledStaking.sol";
 import "./NXMToken.sol";
-import "./external/IERC1132.sol";
 
-contract TokenController is IERC1132, Iupgradable {
+contract TokenController is Iupgradable {
   using SafeMath for uint256;
 
-  event Burned(address indexed member, bytes32 lockedUnder, uint256 amount);
+  struct LockToken {
+    uint256 amount;
+    uint256 validity;
+    bool claimed;
+  }
+
+  struct CoverInfo {
+    uint16 claimCount;
+    bool hasOpenClaim;
+    bool hasAcceptedClaim;
+    // note: still 224 bits available here, can be used later
+  }
+
+  // _of => reason
+  mapping(address => bytes32[]) public lockReason;
+
+  // _of => reason => LockToken
+  mapping(address => mapping(bytes32 => LockToken)) public locked;
 
   NXMToken public token;
   IPooledStaking public pooledStaking;
-  uint public minCALockTime = uint(30).mul(1 days);
+
+  uint public minCALockTime;
+  uint public claimSubmissionGracePeriod;
+
+  // coverId => CoverInfo
+  mapping(uint => CoverInfo) public coverInfo;
```

Comment:
this cover info being in tokencontroller doesn't make sense in terms of logical location, there's no other proxy contract where it would make more sense though, only some other data contract where you hold cover data

Reply:
note on off-band discussion: 
we know that the CoverInfo mapping only gets populated for covers that have at least 1 claim, for others the fields are zeroed out.

Therefore in case we need to migrate this data structure to a different contract, we would only need to migrate the covers that have at least 1 claim which are as of today few in number so it would be feasible in terms of costs